### PR TITLE
Add solution for Convert Binary Number in a Linked List to Integer

### LINKED ISSUE #33: Convert Binary Number in a Linked List to Integer
## Issue Overview
Implement a solution for the "Convert Binary Number in a Linked List to Integer" LeetCode problem to practice linked list traversal and binary conversion in Rust.

## Feature Details
Create a Rust implementation for the problem: https://leetcode.com/problems/convert-binary-number-in-a-linked-list-to-integer/

Problem summary:
Given the `head` of a singly linked list where each node contains a binary digit (0 or 1), return the decimal value of the number in the linked list.

Constraints:
- The Linked List is not empty.
- Number of nodes will not exceed 30.
- Each node's value is either 0 or 1.

The solution should:
- Implement a linked list structure in Rust
- Demonstrate traversal of linked lists
- Show binary to decimal conversion techniques
- Include proper error handling
- Feature comprehensive tests

## Expected Outcome
A well-documented Rust implementation that correctly converts a binary number represented as a linked list to its decimal value with clean, efficient code, proper error handling, and thorough test coverage. This will also introduce linked list handling in our Rust code repository.

### PR DESCRIPTION
**Pull request by zmrdltl**
Related Issues
closes #33 

PR Description
Adds a solution for the LeetCode "Convert Binary Number in a Linked List to Integer" problem.

Implementation details:
- Convert a binary number, represented as a linked list, to an integer value
- Each node in the linked list contains a single binary digit (0 or 1)
- Process the linked list with a simple iterative approach:
  - Start with result = 0
  - For each node: result = result * 2 + node.val
- Validate input and return appropriate error for empty linked lists
- O(n) time complexity where n is the length of the linked list
- O(1) space complexity as we only use a single variable for computation

The solution efficiently traverses the linked list once, building the decimal
value as it goes. This approach mimics how binary-to-decimal conversion works
mathematically, with each new digit shifting the existing value left (×2) and
then adding the new digit.

Test cases cover both standard scenarios and edge cases like empty lists.

Special Notes
The implementation demonstrates how to handle Option<Box<T>> patterns in Rust
and effectively process linked data structures.

### CODE REVIEW

File: src/encoding_decoding/convert_binary_number_in_a_linked_list_to_integer.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+pub fn get_decimal_value(head: Option<Box<ListNode>>) -> Result<i32, &'static str> {
+    if head.is_none() {
+        return Err("Head is None");
+    }
+
+    let mut current = head;
+    let mut result = 0;
+
+    while let Some(node) = current {
+        result = result * 2 + node.val;
```
Comment (kimhanbeom) [inline src/encoding_decoding/convert_binary_number_in_a_linked_list_to_integer.rs:45]: 각 `ListNode`의 `value`는 bit를 의미하므로 이 부분 또한 bit 연산으로 변경하는 것은 어떨까요?

**Review by kimhanbeom — CHANGES_REQUESTED** [review]

**Review by kimhanbeom — APPROVED** [review]
